crates/trigger-http/src/server.rs

```diff
@@ -138,12 +150,18 @@ impl<F: RuntimeFactors> HttpServer<F> {
 
     /// Serve incoming requests over the provided [`TcpListener`].
     pub async fn serve(self: Arc<Self>) -> anyhow::Result<()> {
-        let listener = TcpListener::bind(self.listen_addr).await.with_context(|| {
-            format!(
-                "Unable to listen on {listen_addr}",
-                listen_addr = self.listen_addr
-            )
-        })?;
+        let listener: TcpListener = if self.find_free_port {
+            self.search_for_free_port().await?
+        } else {
+            TcpListener::bind(self.listen_addr).await.map_err(|err| {
+                if err.kind() == ErrorKind::AddrInUse {
+                    anyhow::anyhow!("Unable to listen on {}. To have Spin search for a free port, use the --find-free-port option.", self.listen_addr)
```

Comment:
It occurs to me that since we now know the kind of error we can reword this to e.g.

```suggestion
                    anyhow::anyhow!("{} is already in use. To have Spin search for a free port, use the --find-free-port option.", self.listen_addr)
```